provider/datagen/src/lib.rs

```diff
@@ -671,12 +671,13 @@ fn test_keys() {
 #[cfg(feature = "legacy_api")]
 fn test_keys_from_file() {
     #![allow(deprecated)]
+
+    let mut buffer = std::env::var("CARGO_MANIFEST_DIR")
+        .or(core::env!("CARGO_MANIFEST_DIR"))
```

Comment:
the types don't work out here. Once you add the conversions this expression will be complex enough to factor it out

Reply:
yeah I was waiting on a final call on what we're supposed to do here before fixing everything